server/src/main/java/com/objectcomputing/checkins/services/kudos/KudosServicesImpl.java

```diff
@@ -0,0 +1,297 @@
+package com.objectcomputing.checkins.services.kudos;
+
+import com.objectcomputing.checkins.exceptions.BadArgException;
+import com.objectcomputing.checkins.exceptions.NotFoundException;
+import com.objectcomputing.checkins.exceptions.PermissionException;
+import com.objectcomputing.checkins.services.kudos.kudos_recipient.KudosRecipient;
+import com.objectcomputing.checkins.services.kudos.kudos_recipient.KudosRecipientRepository;
+import com.objectcomputing.checkins.services.kudos.kudos_recipient.KudosRecipientServices;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRetrievalServices;
+import com.objectcomputing.checkins.services.memberprofile.currentuser.CurrentUserServices;
+import com.objectcomputing.checkins.services.team.Team;
+import com.objectcomputing.checkins.services.team.TeamRepository;
+import com.objectcomputing.checkins.util.Util;
+import io.micronaut.core.annotation.Nullable;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Singleton
+public class KudosServicesImpl implements KudosServices {
+
+    private final KudosRepository kudosRepository;
+    private final KudosRecipientServices kudosRecipientServices;
+    private final KudosRecipientRepository kudosRecipientRepository;
+    private final TeamRepository teamRepository;
+    private final MemberProfileRetrievalServices memberProfileRetrievalServices;
+    private final CurrentUserServices currentUserServices;
+
+    public static final Logger LOG = LoggerFactory.getLogger(KudosServicesImpl.class);
+
+    public KudosServicesImpl(KudosRepository kudosRepository,
+                             KudosRecipientServices kudosRecipientServices,
+                             KudosRecipientRepository kudosRecipientRepository,
+                             TeamRepository teamRepository,
+                             MemberProfileRetrievalServices memberProfileRetrievalServices,
+                             CurrentUserServices currentUserServices) {
+        this.kudosRepository = kudosRepository;
+        this.kudosRecipientServices = kudosRecipientServices;
+        this.kudosRecipientRepository = kudosRecipientRepository;
+        this.teamRepository = teamRepository;
+        this.memberProfileRetrievalServices = memberProfileRetrievalServices;
+        this.currentUserServices = currentUserServices;
+    }
+
+    @Override
+    public Kudos save(KudosCreateDTO kudosDTO) {
+
+        memberProfileRetrievalServices.getById(kudosDTO.getSenderId()).orElseThrow(() -> {
+            throw new BadArgException("Kudos sender %s does not exist");
+        });
```

Comment:
```suggestion
        memberProfileRetrievalServices.getById(kudosDTO.getSenderId()).orElseThrow(() ->
            new BadArgException("Kudos sender %s does not exist")
        );
```

`orElseThrow` throws the exception returned by the lambda, the lambda does not itself throw the exception (although the end result is probably the same) 😉